crates/driver/src/domain/quote.rs

```diff
@@ -62,15 +60,40 @@ pub struct Order {
     pub amount: order::TargetAmount,
     pub side: order::Side,
     pub gas_price: eth::EffectiveGasPrice,
+    pub deadline: Deadline,
+}
+
+/// The deadline for computing a quote for an order.
+#[derive(Clone, Copy, Debug, Default)]
+pub struct Deadline(chrono::DateTime<chrono::Utc>);
```

Comment:
It's a bit unfortunate that we end up with duplicated type implementations. What do you think about having a single shared `Deadline` type but giving it some `PhantomData` so we can have something like `Deadline<Solving>` and `Deadline<Quoting>`.
This could also be used for an execution deadline.

Reply:
Ehhh not sure. It works but it's not pretty. There's not _that much_ duplicated code.

OTOH since quoting is just a form of solving, I don't think I'd disagree with just using `solution::Deadline` here.

Reply:
> I don't think I'd disagree with just using solution::Deadline here.

One reason I'm not super fond of this, is that it prevents us from using different "timeout buffers" for both solving and quoting (which should be different, the post-quote work is **much** quicker - not involving any external API roundtrip). If we did this, we would leak the choice of "time buffer" to use outside of the `solution` module which I thought wasn't ideal.

I think we can get rid of the duplicated validity check by moving it to when we render a  `SolverTimeout`, then the `Deadline` is as much boiler-plate as a marker type.

Reply:
Alright, so I ended up settling on something like this: c9409ae3fb16acf921bfcb507bf0865add9728ae

Specifically, instead of `SolveTimeout::new(deadline, now)`, we now have `deadline.timeout(now)`. This change makes is so that the `time_buffer` details are tied to the specific deadlines (we have `auction::Deadline::time_buffer` and `quote::Deadline::time_buffer`) and aren't "leaked" into the `solution` module. Also, to reduce boilerplate, `Deadline`'s aren't checked against `now`, but instead this only happens when rendering a `SolverTimeout`. This is nice because:
- We only check once that the timeout is reasonable when actually "using" a deadline, making it so we don't have any copied logic, and just boilerplate (conversion trait implementations) and deadline-specific logic (different timeouts and error messages).
- Simplifies code (`now` doesn't need to be passed in as many places)

Let me know what you think.

Reply:
Niiice.